plugins/internal/telemetry/http_transport.go

```diff
@@ -0,0 +1,121 @@
+package telemetry
+
+import (
+	"io"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// Refer OpenTelemetry Semantic Conventions for HTTP Client.
+// https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#http-client
+const (
+	metricClientDuration     = "http.client.duration"
+	metricClientRequestSize  = "http.client.request.size"
+	metricClientResponseSize = "http.client.response.size"
+
+	attributeServerPort         = "server.port"
+	attributeServerAddress      = "server.address"
+	attributeRequestMethod      = "http.request.method"
+	attributeResponseStatusCode = "http.response.status_code"
+)
+
+func NewHTTPTransport(baseTransport http.RoundTripper) http.RoundTripper {
+	if _, ok := baseTransport.(*httpTransport); ok {
+		return baseTransport
+	}
+
+	if baseTransport == nil {
+		baseTransport = http.DefaultTransport
+	}
+
+	icl := &httpTransport{roundTripper: baseTransport}
+	icl.createMeasures(otel.Meter("github.com/goto/meteor/plugins/internal/telemetry"))
+
+	return icl
+}
+
+type httpTransport struct {
+	roundTripper http.RoundTripper
+
+	counters       map[string]metric.Int64Counter
+	valueRecorders map[string]metric.Float64Histogram
+}
+
+func (tr *httpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	ctx := req.Context()
+	startAt := time.Now()
+
+	var bw bodyWrapper
+	if req.Body != nil && req.Body != http.NoBody {
+		bw.ReadCloser = req.Body
+		req.Body = &bw
+	}
+
+	attribs := metric.WithAttributes(
+		attribute.String(attributeRequestMethod, req.Method),
+		attribute.String(attributeServerAddress, req.URL.Hostname()),
+		attribute.String(attributeServerPort, req.URL.Port()),
+	)
+
+	tr.counters[metricClientRequestSize].Add(ctx, bw.read.Load(), attribs)
+
+	resp, err := tr.roundTripper.RoundTrip(req)
+	elapsedTime := float64(time.Since(startAt)) / float64(time.Millisecond)
+	tr.valueRecorders[metricClientDuration].Record(ctx, elapsedTime, attribs)
+
+	if err == nil {
+		tr.counters[metricClientResponseSize].Add(ctx, resp.ContentLength, attribs,
+			metric.WithAttributes(attribute.Int(attributeResponseStatusCode, resp.StatusCode)))
+	}
+
+	return resp, err
+}
+
+func (tr *httpTransport) createMeasures(meter metric.Meter) {
+	tr.counters = make(map[string]metric.Int64Counter)
+	tr.valueRecorders = make(map[string]metric.Float64Histogram)
+
+	requestBytesCounter, err := meter.Int64Counter(metricClientRequestSize)
+	handleErr(err)
+
+	responseBytesCounter, err := meter.Int64Counter(metricClientResponseSize)
+	handleErr(err)
+
+	clientLatencyMeasure, err := meter.Float64Histogram(metricClientDuration)
+	handleErr(err)
+
+	tr.counters[metricClientRequestSize] = requestBytesCounter
+	tr.counters[metricClientResponseSize] = responseBytesCounter
+	tr.valueRecorders[metricClientDuration] = clientLatencyMeasure
+}
+
+func handleErr(err error) {
+	if err != nil {
+		otel.Handle(err)
+	}
+}
```

Comment:
Not a review comment for this specific change but should we consistently apply this for all instances where we create otel meters?

Reply:
Essentially, we are only logging the error and proceeding with execution even if instrumentation fails for unknown and unexpected reason.